src/csharp/Microsoft.Spark/Interop/Ipc/SerDe.cs

```diff
@@ -183,6 +182,32 @@ public static byte[] ReadBytes(Stream s, int length)
             return buffer;
         }
 
+        internal static void ReadBytes(Stream s, byte[] buffer, int length)
+        {
+            if (length < 0)
+            {
+                throw new ArgumentOutOfRangeException(nameof(length), length, "length can't be negative.");
+            }
+            if (length == 0)
+            {
+                return;
+            }
+
+            int bytesRead;
```

Comment:
I find the design of `TryReadBytes` confusing:

- it returns false when it can't read any bytes (it's ok, it's a Try method)
- it throws an exception when the read is incomplete `totalBytesRead < messageLength` (I would not expect a Try method to throw)

IMHO it should throw in both cases or return false in both. Introducing a new method that always throws when something is wrong seemed like a better approach to me. But it's only a opinion, I can change the code if you want.

Reply:
> I would not expect a Try method to throw

Try* methods often throw.  The Try pattern isn't about eliminating all exceptions: it's explicitly about picking one category of exceptions that's likely to happen and turning just that one category into a Boolean result.

Reply:
What was the resolution here? IMO it would be nice if we could just have a single pattern in the code.

Reply:
I removed the method and reused `TryReadBytes`